spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/RdbmsTaskDefinitionRepository.java

```diff
@@ -0,0 +1,297 @@
+	}
+
+	@Override
+	public <S extends TaskDefinition> S save(S definition) {
+		Assert.notNull(definition, "definition must not be null");
+		if (exists(definition.getName())) {
+			throw new DuplicateTaskException(
+					String.format("Cannot register task %s because another one has already " +
+									"been registered with the same name",
+							definition.getName()));
+		}
+
+		Object[] insertParameters = new Object[]{definition.getName(), definition.getDslText()};
+		jdbcTemplate.update(prefixSql(SAVE_TASK_DEFINITION), insertParameters,
+				new int[]{Types.VARCHAR, Types.CLOB});
+
+		return definition;
+	}
+
+	@Override
+	public TaskDefinition findOne(String taskName) {
+		Assert.hasText(taskName, "taskName must not be empty nor null");
+		try {
+			return jdbcTemplate.queryForObject(prefixSql(FIND_TASK_DEFINITION_BY_NAME),
+					new TaskDefinitionRowMapper(),
+					taskName);
+		}
+		catch (EmptyResultDataAccessException e) {
+			return null;
+		}
+	}
+
+	@Override
+	public boolean exists(String taskName) {
+		Assert.hasText(taskName, "taskName must not be empty nor null");
+		boolean result;
+		try {
+			result = (jdbcTemplate.queryForObject( prefixSql(TASK_DEFINITION_COUNT_BY_NAME),
+					new Object[]{taskName}, Long.class) > 0) ? true : false;
+		}
+		catch (EmptyResultDataAccessException e) {
+			result = false;
+		}
+		return result;
+	}
+
+	@Override
+	public Iterable<TaskDefinition> findAll() {
+		return Collections.unmodifiableCollection(
+				jdbcTemplate.query(prefixSql(FIND_ALL_DEFINITIONS), new TaskDefinitionRowMapper()));
+	}
+
+	@Override
+	public Iterable<TaskDefinition> findAll(Iterable<String> taskNames) {
+		Assert.notNull(taskNames, "taskNames must not be null");
+		List<String> listOfNames = new ArrayList<String>();
+		for (String name : taskNames) {
+			listOfNames.add(name);
+		}
+
+		MapSqlParameterSource namedParameters = new MapSqlParameterSource();
+		namedParameters.addValue(LIST_OF_NAMES, listOfNames);
+
+		return Collections.unmodifiableCollection(
+				namedParameterJdbcTemplate.query(prefixSql(FIND_DEFINITIONS_WHERE_IN_CLAUSE),
+				namedParameters, new TaskDefinitionRowMapper()));
+	}
+
+	@Override
+	public long count() {
+		try {
+			return jdbcTemplate.queryForObject(
+					prefixSql(TASK_DEFINITION_COUNT), new Object[]{}, Long.class);
+		}
+		catch (EmptyResultDataAccessException e) {
+			return 0;
+		}
+	}
+
+	@Override
+	public void delete(String taskName) {
+		Assert.hasText(taskName, "taskName must not be empty nor null");
+		jdbcTemplate.update(prefixSql(DELETE_FROM_TASK_DEFINITION_BY_NAME), taskName);
+	}
+
+	@Override
+	public void delete(TaskDefinition definition) {
+		Assert.notNull(definition, "definition must not null");
+		delete(definition.getName());
+	}
+
+	@Override
+	public void delete(Iterable<? extends TaskDefinition> definitions) {
+		Assert.notNull(definitions, "definitions must not null");
+		for (TaskDefinition definition : definitions) {
+			delete(definition);
+		}
+	}
```

Comment:
In most of these methods, can we do the call to `prefixSql` once and cache it so that we don't need to do that string manipulation for every call?